course_catalog/tasks_test.py

```diff
@@ -104,19 +104,43 @@ def setup_s3(settings):
     conn.create_bucket(Bucket=settings.OCW_LEARNING_COURSE_BUCKET_NAME)
 
 
+@mock_s3
 def test_get_mitx_data_valid(
     settings, access_token, get_test_data, mock_course_index_functions
 ):
     """
     Test that mitx sync task successfully creates database objects
     """
+    setup_s3(settings)
     get_edx_data()
     assert Course.objects.count() == 1
     assert CoursePrice.objects.count() == 2
     assert CourseInstructor.objects.count() == 2
     assert CourseTopic.objects.count() == 1
 
 
+@mock_s3
+def test_get_mitx_data_saves_json(
+    settings, mocker, access_token, get_test_data, mock_course_index_functions
+):
+    """
+    Test that mitx sync task successfully saves edx API response results in S3
+    """
+    setup_s3(settings)
+    get_edx_data()
+    conn = boto3.client(
+        "s3",
+        aws_access_key_id=settings.OCW_LEARNING_COURSE_BUCKET_NAME,
+        aws_secret_access_key=settings.OCW_LEARNING_COURSE_ACCESS_KEY,
+    )
+    json_is_uploaded = conn.list_objects(
+        Bucket=settings.OCW_LEARNING_COURSE_BUCKET_NAME, Prefix="edx_courses.json"
+    )
+    # check that pub_object call to create edx_courses.json succeeded
+    assert json_is_uploaded
```

Comment:
This will be `True` even if the file doesn't exist.  Can you retrieve the body of the object and verify it contains expected course results.

```python
conn.list_objects(
    Bucket=settings.OCW_LEARNING_COURSE_BUCKET_NAME, 
    Prefix="totally_fake_filename.json"
)

{'EncodingType': 'url',
 'IsTruncated': False,
 'Marker': '',
 'MaxKeys': 1000,
 'Name': 'open-learning-course-data-ci',
 'Prefix': 'totally_fake_filename.json',
 'ResponseMetadata': {'HTTPHeaders': {'content-type': 'application/xml',
   'date': 'Fri, 15 Feb 2019 21:53:39 GMT',
   'server': 'AmazonS3',
   'transfer-encoding': 'chunked',
   'x-amz-bucket-region': 'us-east-1',
   'x-amz-id-2': 'rBWLBWdwFYrqJETvlgmQMi2f8hfYd8WVcyCnGActq7WPcyoCrcu2Epm3Wv3/t4vPPV1Y/np7qB0=',
   'x-amz-request-id': '5BEB8BB6BDF74530'},
  'HTTPStatusCode': 200,
  'HostId': 'rBWLBWdwFYrqJETvlgmQMi2f8hfYd8WVcyCnGActq7WPcyoCrcu2Epm3Wv3/t4vPPV1Y/np7qB0=',
  'RequestId': '5BEB8BB6BDF74530',
  'RetryAttempts': 0}}
```

Reply:
Good point. Updated the unit test.